app/components/mode-toggle.tsx

```diff
@@ -0,0 +1,38 @@
+"use client";
+
+import { Moon, Sun } from "lucide-react";
+import { useTheme } from "next-themes";
+import { Button } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+
+export function ModeToggle() {
+  const { setTheme } = useTheme();
+
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button variant="outline" size="icon">
+          <Sun className="h-5 w-5 dark:hidden" />
+          <Moon className="absolute h-5 w-5 hidden dark:block" />
```

Comment:
this is unnecessary here
```suggestion
          <Moon className="h-5 w-5 hidden dark:block" />
```